internal/db/db.go

```diff
@@ -143,115 +144,253 @@ func (tx *holdTx) FindOne(modelID uuid.UUID, matcher Matcher) (rec Record, err e
 	return
 }
 
-func (tx holdTx) FindMany(modelID uuid.UUID, matcher Matcher) []Record {
-	mi := tx.h.IterMatches(modelID, matcher)
-	hits := []Record{}
-	for val, ok := mi.Next(); ok; val, ok = mi.Next() {
-		hits = append(hits, val)
+func (tx *holdTx) FindMany(modelID uuid.UUID, matcher Matcher) (recs []Record, err error) {
+	recs, err = tx.h.FindMany(modelID, matcher)
+	return
+}
+
+func (tx *holdTx) Insert(rec Record) error {
+	tx.ensureWrite()
+	h, err := tx.h.Insert(rec)
+	if err != nil {
+		return err
 	}
-	return hits
+	tx.h = h
+	return nil
 }
 
-func (tx *holdTx) Insert(rec Record) {
+func (tx *holdTx) Update(r Record, fields map[string]interface{}) (Record, error) {
 	tx.ensureWrite()
-	tx.h = tx.h.Insert(rec)
+	//Need to load a model for the record
+	m, _ := tx.GetModel(r.Type())
+	newRec := r.DeepCopy(m)
```

Comment:
curious why we have to pass the model into DeepCopy. I think that's worth fixing before we merge this.